### PR TITLE
feat: Added Azure.VMSS.AvailabilityZone

### LINKED ISSUE #2902: [RULE] Deploy VMSS across availability zones with VMSS Flex
### Existing rule

_No response_

### Suggested rule

As per [APRL recommendation](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/azure-resources/Compute/virtualMachineScaleSets/#deploy-vmss-across-availability-zones-with-vmss-flex), when creating VMSS, implement availability zones as a protection measure for your applications and data against the rare event of datacenter failure.

Resource Type **_microsoft.compute/virtualmachinescalesets_**
Property with expected value **_array_length(zones) > 1_**

### Pillar

Reliability

### Additional context

[Reliability Hub](https://learn.microsoft.com/en-us/azure/reliability/reliability-virtual-machine-scale-sets?tabs=graph-4%2Cgraph-1%2Cgraph-2%2Cgraph-3%2Cgraph-5%2Cgraph-6%2Cportal#-deploy-virtual-machine-scale-sets-across-availability-zones-with-virtual-machine-scale-sets-flex)
[Create a Virtual Machine Scale Set that uses Availability Zones](https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-use-availability-zones?tabs=cli-1%2Cportal-2)
[Update scale set to add availability zones](https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-use-availability-zones?tabs=cli-1%2Cportal-2#update-scale-set-to-add-availability-zones)

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2902

Azure.VMSS.AvailabilityZone.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
Ready for review @BernieWhite.

File: docs/en/rules/Azure.VMSS.AvailabilityZone.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## DESCRIPTION
+
+Virtual machine scale set supports the use of availability zones to provide zone redundancy. Zone redundancy enhances the resiliency and high availability of the virtual machine scale set by deploying instances across data centers in physically separated zones.
+
+Virtual machine scale set utilizes scale rules, and as the set scales, it creates instances within the zones it is configured to use. If the set is configured to use only Zone 1, all new instances will be created in Zone 1. However, if the set is configured to use all three zones (Zone 1, Zone 2, and Zone 3), new instances will be distributed across these zones as it scales, ensuring balanced distribution and improved resilience.
+
+Using availability zones is suitable for stateless workloads.
+
+See documentation references below for additional limitations and important information.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.VMSS.AvailabilityZone.md:23]:
```suggestion
Virtual machine scale sets (VMSS) supports the use of availability zones to provide redundancy across physically separated zones.
When new instances are added to the VMSS as it scales, these instances are deployed across any zones that are configured.

For example:

- If the VMSS is configured to use only Zone 1, all new instances will be created in Zone 1.
- If the VMSS is configured to use three zones (Zone 1, Zone 2, and Zone 3), new instances will be distributed across these zones.

Also note:

- Using availability zones is suitable for stateless workloads.
- During scale-in operations, the number of instances in each zone may not be evenly distributed for all zones.

See documentation references below for additional limitations and important information.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Great work @BenjaminEngeset. Please see the additional suggestions below.

**Comment by BenjaminEngeset** [comment]
Hi @BernieWhite,

Thanks for the great feedback. I have adjusted accordingly now. Plesase check.

**Review by BernieWhite — APPROVED** [review]
Thanks @BenjaminEngeset. All good to merge.

**Pull request merged**